types/index.esm.d.ts

```diff
@@ -2479,12 +2476,16 @@ export interface TooltipItem {
 	element: Element;
 }
 
-export interface PluginOptions {
+export interface PluginOptionsByType {
 	filler: FillerOptions;
 	legend: LegendOptions;
 	title: TitleOptions;
 	tooltip: TooltipOptions;
 }
+export interface PluginChartOptions {
```

Comment:
Out of the scope of this PR: I'm wondering if all the `*ChartOptions` (e.g. `PluginChartOptions`, `ScaleChartOptions`, etc.) should be exported (thus public). These are kind of helpers to share structure between defaults, controllers, and config, and it doesn't seem to make sense to be able to augment their definitions.

Reply:
I think they could not be exported since I don't expect a user to change them. Let's add it in a separate PR.